runtime/fundamentals/ts_support.md

```diff
@@ -0,0 +1,337 @@
+---
+title: "TypeScript support"
+oldUrl:
+  - /runtime/manual/advanced/typescript/
+  - /runtime/manual/typescript/
+  - /runtime/manual/typescript/overview/
+  - /runtime/manual/getting_started/typescript/
+  - /manual/advanced/typescript
+  - /runtime/manual/advanced/typescript/overview/
+  - /runtime/manual/advanced/typescript/types/
+---
+
+TypeScript is a first class language in Deno, just like JavaScript or
+WebAssembly, so you can run or import TypeScript without installing anything
+more than the Deno CLI.
+
+## The TS Compiler
+
+Deno has a built-in TypeScript compiler, which is used to compile TypeScript
+code to JavaScript. No extra installation or config is required making your
+development process simpler.
+
+### Automatic conversion and caching
+
+Deno converts TypeScript (as well as TSX and JSX) into JavaScript. It does this
+with a combination of the built in
+[TypeScript compiler](https://github.com/microsoft/TypeScript), and a Rust
+library called [swc](https://swc.rs/). When the code has been type checked and
+transformed, it is stored in a cache, ready for the next run without the need to
+convert it from its source to JavaScript again.
+
+You can see this cache location by running `deno info`:
+
+```shell
+$ deno info
+DENO_DIR location: "/path/to/cache/deno"
+Remote modules cache: "/path/to/cache/deno/deps"
+TypeScript compiler cache: "/path/to/cache/deno/gen"
+```
+
+If you were to look in the cache, you would see a directory structure that
+mimics your source directory structure and individual `.js` and `.meta` files.
+The `.js` file is the transformed source file while the `.meta` file contains
+meta data we want to cache about the file. The metadata contains a _hash_ of the
+source module that helps Deno to manage cache invalidation. You might also see a
+`.buildinfo` file, a TypeScript compiler incremental build information file,
+used to help speed up type checking.
```

Comment:
Is this relevant for Deno users? I feel like we're going deep into implementation details here. For users it doesn't matter if Deno transpiles it internally to JS or if we do it any other way. The main takeaway is that Deno runs TS natively.